src/main/java/org/gosulang/gradle/GosuPlugin.java

```diff
@@ -29,26 +29,26 @@ public void apply(Project project) {
    * Ensures that the runtime dependency on gosu-core is included the testRuntime's classpath
    */
   private void refreshTestRuntimeClasspath( final Project project ) {
-    final JavaPluginConvention pluginConvention = project.getConvention().getPlugin(JavaPluginConvention.class);
+    final JavaPluginExtension pluginExtension = project.getExtensions().getByType(JavaPluginExtension.class);
     GosuRuntime gosuRuntime = project.getExtensions().getByType(GosuRuntime.class);
 
-    SourceSet main = pluginConvention.getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME);
-    SourceSet test = pluginConvention.getSourceSets().getByName(SourceSet.TEST_SOURCE_SET_NAME);
+    SourceSet main = pluginExtension.getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME);
+    SourceSet test = pluginExtension.getSourceSets().getByName(SourceSet.TEST_SOURCE_SET_NAME);
 
     test.setRuntimeClasspath(project.files(
         test.getOutput(),
         main.getOutput(),
-        project.getConfigurations().getByName(JavaPlugin.TEST_RUNTIME_CONFIGURATION_NAME),
-        gosuRuntime.inferGosuClasspath(project.getConfigurations().getByName(JavaPlugin.TEST_COMPILE_CONFIGURATION_NAME))));
+        project.getConfigurations().getByName(JavaPlugin.TEST_RUNTIME_ONLY_CONFIGURATION_NAME),
```

Comment:
Consider using `testRuntimeClasspath` that automatically combines testRuntime and testImplementation,
see https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_plugin_and_dependency_management (scroll down to see the diagrams of configuration dependencies)

Also there might be an error when you try to resolve TEST_RUNTIME_CONFIGURATION_NAME as it is not resolvable by default, while testRuntimeClasspath is resolvable. Example of this behavior https://stackoverflow.com/questions/47910578/not-able-to-copy-configurations-dependencies-after-upgrading-gradle-plugin-for-a

Reply:
Updated the configuration. Thank you